### DISCUSSION TITLE
CallPeaks() function on multiple fragments.tsv.gz files

### DISCUSSION THREAD

**Discussion by hongruhu** [stuart-lab/signac#1420] [Q&A] [answered]
Hi I was wondering if I could use the `CallPeaks()` function on multiple fragments.tsv.gz files' paths?

For example I have 3 files and they have different fragments, `A.tsv.gz`, `B.tsv.gz`, and `C.tsv.gz`:
```
        1: chr1  3000137  3000371 TGCTTCGTCCATTGAG-1  1
        2: chr1  3000147  3000361 AGGCGAAAGCAATAAC-1  6
        3: chr1  3000189  3000366 TCAAAGCAGTCGTACT-1 10
```

```
        1: chr1  3000206  3000282 AGGCCCAAGTAACATG-1  3
        2: chr1  3000225  3000280 TACTAGGCAATTAGGA-1  1
        3: chr1  3000322  3000587 TTAGCGAAGTCTCGAT-1  2

```
```
        1: chr1  3000035  3000294 AATGTCGCATTGCCTC-1  7
        2: chr1  3000086  3000280 CTAGGGCGTACAAGCG-1 20
        3: chr1  3000149  3000190 TCGCAGGCACTGTCAA-1  3
```
Is there anyway to get the cell x peak matrix for each of the 3 datasets with the same features(peaks)

Thanks!

**Comment by timoast** [comment, marked as answer]
The `CallPeaks()` function can use a list of fragment files, or if there are multiple fragment files linked to the object they will all be used. Similarly for `FeatureMatrix()`, all the fragment files linked to the object are used

**Reply by AmosFong1** [reply]
@timoast If there are a list of fragment files, can you still use the group.by argument to run MACS2 on subsets of cells?
